.ci/lib.sh

```diff
@@ -512,3 +512,244 @@ print_environment() {
+	pr="$ghprbPullId"
+
+	local result=$(is_label_set "$repo" "$pr" "$label")
+	if [ "$result" -eq 1 ]; then
+		local_info "label '$label' found"
+		echo "1"
+		return 0
+	fi
+
+	local_info "label '$label' not found"
+	echo "0"
+	return 0
+
+}
+
+# We don't want to have a main fucntion to trigger unit tests in this library,
+# so prepend these tests with `lib_`, so they can be easily called elsewhere
+testIsLabelSet() {
+	local result=""
+
+	result=$(is_label_set "" "" "")
+	assertEquals "0" "$result"
+
+	result=$(is_label_set "repo" "" "")
+	assertEquals "0" "$result"
+
+	result=$(is_label_set "repo" "pr" "")
+	assertEquals "0" "$result"
+
+	result=$(is_label_set "" "pr" "label")
+	assertEquals "0" "$result"
+
+	result=$(is_label_set "repo" "" "label")
+	assertEquals "0" "$result"
+
+	result=$(is_label_set "repo" "pr" "label")
+	assertEquals "1" "$result"
+}
+
+testCheckLabel() {
+	local result=""
+
+	result=$(check_label "")
+	assertEquals "0" "$result"
+
+	result=$(unset ghprbGhRepository; check_label "label")
+	assertEquals "0" "$result"
+
+	result=$(unset ghprbPullId; check_label "label")
+	assertEquals "0" "$result"
+
+	result=$(unset ghprbGhRepository ghprbPullId; check_label "label")
+	assertEquals "0" "$result"
+
+	# Pretend label not found
+	result=$(is_label_set() { echo "0"; return 0; }; \
+	                ghprbGhRepository="repo"; \
+	                ghprbPullId=123; \
+	                check_label "label")
+	assertEquals "0" "$result"
+
+	# Pretend label found
+	result=$(is_label_set() { echo "1"; return 0; }; \
+	                ghprbGhRepository="repo"; \
+	                ghprbPullId=123; \
+	                check_label "label")
+	assertEquals "1" "$result"
+}
+
+# Run our self tests. Tests are written using the
+# github.com/kward/shunit2 library, and are encoded into functions starting
+# with the string 'test'.
+self_test() {
+	local shunit2_path="github.com/kward/shunit2"
+	local_info "Running self tests"
+
+	local_info "Go get unit test framework from ${shunit2_path}"
+	go get -d "${shunit2_path}" || true
+	local_info "Run the unit tests"
+	# Sourcing the `shunit2` file automatically runs the unit tests in this file.
+	. "${GOPATH}/src/${shunit2_path}/shunit2"
+	# shunit2 call does not return - it exits with its return code.
+}
+
+help()
+{
+	script_name="${0##*/}"
+	cat <<EOF
+Usage: ${script_name} [test]
+
+Passing the argument 'test' to this script will cause it to only
+run its self tests.
+EOF
+
+	exit 0
+}
+
+main() {
+	# Some of our sub-funcs return their results on stdout, but we also want them to be
+	# able to log INFO messages. But, we don't want those going to stderr, as that may
+	# be seen by some CIs as an actual error. Create another file descriptor, mapped
+	# back to stdout, for us to send INFO messages to...
+	exec 5>&1
+
+	if [ "$1" == "test" ]; then
+		self_test
+		# self_test func does not return
+	fi
+
+	[ $# -gt 0 ] && help
+}
+
+[ "$0" == "$BASH_SOURCE" ] && main "$@" || true
```

Comment:
I think this would be inverse, this is true _when_ `source`d. (Did this work?!)

Reply:
I'm confused. Here is a little test script I made:
```bash
echo "\$0: $0"
echo "\$BASH_SOURCE: $BASH_SOURCE"
[ "$0" == "$BASH_SOURCE" ] && echo "not sourced"
```
and the results of me running it not sourced:
```
$ ./test.sh
$0: ./test.sh
$BASH_SOURCE: ./test.sh
not sourced
```
and sourced:
```
$ . test.sh
$0: -bash
$BASH_SOURCE: test.sh
```

Reply:
Ah no, okay, had a misunderstanding about how `$BASH_SOURCE` works. Thanks for checking!